sale_procurement_group_by_line/model/sale.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2013-2014 Camptocamp SA - Guewen Baconnier
+# © 2016-20 ForgeFlow S.L. (https://www.forgeflow.com)
+# © 2016 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.float_utils import float_compare
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.model
+    def _prepare_procurement_group_by_line(self, line):
+        """ Hook to be able to use line data on procurement group """
+        return {"name": line.order_id.name}
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def _get_procurement_group_key(self):
+        """Return a key with priority to be used to regroup lines in multiple
+        procurement groups
+
+        """
+        return 8, self.order_id.id
+
+    def _action_launch_stock_rule(self, previous_product_uom_qty=False):
```

Comment:
This method looks mostly like a copy-paste of https://github.com/odoo/odoo/blob/ca7d5701c6a46f7781bdc9af9f2d26cca880eb93/addons/sale_stock/models/sale_order.py#L560, can you update this to reflect changes that happened in upstream?
For instance here we are missing https://github.com/odoo/odoo/blob/ca7d5701c6a46f7781bdc9af9f2d26cca880eb93/addons/sale_stock/models/sale_order.py#L569